src/Orleans/GrainDirectory/IGrainRegistrar.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime;
+
+namespace Orleans.GrainDirectory
+{
+    /// <summary>
+    /// A grain registrar takes responsibility of coordinating the registration of a grains,
+    /// possibly involving multiple clusters. 
+    /// The grain registrar is called only on the silo that is the owner for that grain.
+    /// </summary>
+    interface IGrainRegistrar
```

Comment:
Same comment here: if this is a remotable (RPC-able) interface our convention is to mark it as IAddressable, or ISystemTarget in this case.